### DISCUSSION TITLE
[mini.files] Copy relative/full path of get_fs_entry()

### DISCUSSION THREAD

**Discussion by felpafel** [nvim-mini/mini.nvim#1206] [Ideas]
Hello, recently I switched to `mini.files` and I'm having a good experience.

One common feature file explorers have is the ability to copy relative and full paths, and I think this could be a feature provided _out of the box_ by this plugin.

Yank full path:

```lua
vim.keymap.set('n', 'Y', function()
    local entry = MiniFiles.get_fs_entry()
    if entry then
        vim.fn.setreg('"', entry.path)
        vim.notify('Yanked full path')
    end
end, { buffer = 0 })
```

Yank relative path:

> ':~:.' can be found in the docs `:h filename-modifiers` (`For maximum shortness, use ":~:.".`)
> ':gs?\\?/?' can be found in the docs on examples. Used to normalize the path separator since mini.files uses `'/'` even on Windows.

```lua
vim.keymap.set('n', 'yy', function()
    local entry = MiniFiles.get_fs_entry()
    if entry then
        vim.fn.setreg('"', vim.fn.fnamemodify(entry.path, ':~:.:gs?\\?/?'))
        vim.notify('Yanked relative path')
    end
end, { buffer = 0 })
```

Of course this is only a suggestion, thanks for your time.

https://github.com/user-attachments/assets/4a1aebd9-9cf8-4617-9664-e113678dfd43

**Comment by echasnovski** [comment]
Thanks for the suggestion!

This indeed looks useful, but I'd rather leave this kind of configuration to the user. Mostly because although `Y` and `yy` are relevant candidates, they do not feel unambiguous. The built-in `Y copies from the cursor column to the end of line, which itself might be a useful operation. So my initial instinct is that it should be used for relative path, while `yy` (copy *full* line) for full path.

> ... Used to normalize the path separator since mini.files uses '/' even on Windows.

Yes, and this still is OK because modern-ish Windows supports it as a path separator.
